### PR TITLE
Feature/support post with hesa url

### PR DESCRIPTION
**Pull request by philratcliffe**
### What

Added code to check a POST request and extract the resource_url

### How to review

Unittests:
    * Run the unit tests - python unittests.py

Functional tests:
    * Uncomment the hesa_url = .... line in __init__.py 
    * fire a post request in using test-scripts/post-url.sh
    * Try it with both valid and invalid body content

### CODE REVIEW

File: unittests.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        self.assertEqual(url, self.test_url)
+
+    def test_with_invalid_data(self):
+        """Test with invalid POST body data"""
+        invalid_body = json.dumps(self.invalid_data)
+        http_req = func.HttpRequest(
+            method='post', url='http://somehost.com/somepath', body=invalid_body.encode('ASCII'))
+        with self.assertRaises(OfsMissingUrlError):
+            get_url_from_req(http_req)
+
```
Comment (nshumoogum) [inline unittests.py]: Would it be useful to add a test for when there is en empty body?

In setup include:

`self.empty_body = {}`

File: unittests.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    def test_with_valid_data(self):
+        """Test with valid body data"""
+        valid_body = json.dumps(self.valid_data)
+        http_req = func.HttpRequest(
+            method='post', url='http://somehost.com/somepath', body=valid_body.encode('ASCII'))
+        url = get_url_from_req(http_req)
+        self.assertEqual(url, self.test_url)
+
+    def test_with_invalid_data(self):
+        """Test with invalid POST body data"""
```
Comment (nshumoogum) [inline unittests.py]: To allow one to add further coverage, consider changing test name and comment to something like the following:

```
def test_with_resource_url_missing(self):
    """Test with invalid POST, missing resource_url in body"""
```

File: unittests.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import azure.functions as func
+from IngestRawXmlHttpTrigger.utils import get_url_from_req, OfsMissingUrlError
+
+
+class TestGetUrlFromReq(unittest.TestCase):
+    def setUp(self):
+        self.test_url = 'http://somehost.com/somepath'
+        self.valid_data = {
+            'resource_url': self.test_url}
+        self.invalid_data = {
```
Comment (nshumoogum) [inline unittests.py]: Worth extending test coverage with `empty_body`?

File: unittests.py (hunk 1 above)
Comment (philratcliffe) [reply unittests.py]: Agreed, have added a test with empty JSON object. Also added a TODO to utils.py to revisit this when we have finalised details of POST format.

File: unittests.py (hunk 2 above)
Comment (philratcliffe) [reply unittests.py]: Yes, have renamed to be more specific.

File: unittests.py (hunk 3 above)
Comment (philratcliffe) [reply unittests.py]: Good point. Have added code for this.
